File: Frontend/src/types/Topic.ts
export type TopicStatus = 'OPEN' | 'CLOSED';
export type VotingSessionStatus = 'OPEN' | 'CLOSED';

export interface VotingSession {
  createdAt: string;
  closedAt?: string;
  duration: number;
  status: VotingSessionStatus;
}

export interface Topic {
  id: number;
  title: string;
  description: string;
  status: TopicStatus;
  createdAt: string;
  updatedAt: string;
}

export interface TopicWithSessions {
  topicId: number;
  title: string;
  description: string;
  status: TopicStatus;
  votingSessions: VotingSession[];
}

export interface TopicListResponse {
  data: Topic[];
  total?: number;
}

File: Api/src/main/java/com/sicredi/votacao/domain/model/Topic.java
package com.sicredi.votacao.domain.model;

import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.time.LocalDateTime;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Topic {

    private Long id;
    private String title;
    private String description;
    private TopicStatus status;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    public static Topic create(String title, String description) {
        if (title == null || title.isBlank()) {
            throw new IllegalArgumentException("Título não pode ser vazio");
        }
        Topic topic = new Topic();
        topic.title = title;
        topic.description = description;
        topic.status = TopicStatus.OPEN;
        topic.createdAt = LocalDateTime.now();
        topic.updatedAt = LocalDateTime.now();
        return topic;
    }

}